container/sagemaker/python_service.py

```diff
@@ -168,21 +172,48 @@ def on_post(self, req, res):
                           .decode('utf-8'))
         model_name = data['model_name']
         base_path = data['url']
-        try:
-            msg = self.grpc_client.add_model(model_name, base_path)
-            res.body = msg
-            res.status = falcon.HTTP_200
-        except Exception as e:  # pylint: disable=W0703
-            e = eval(str(e))  # pylint: disable=W0123
-            if e[0] == 409:
-                res.status = falcon.HTTP_409
-            else:
-                res.status = falcon.HTTP_500
-            res.body = e[1].encode('utf-8')
+        if self.validate_model_dir(base_path):
+            try:
+                msg = self.grpc_client.add_model(model_name, base_path)
+
+                # make sure all versions of model is ready before returning the call
+                with self._timeout(seconds=60):
+                    while True:
+                        model_status = self._get_model_status(model_name)
+                        if self._check_all_versions_available(model_status):
+                            break
+                        sleep(1)
+
+                res.body = msg
+                res.status = falcon.HTTP_200
+            except Exception as e:  # pylint: disable=W0703
+                e = eval(str(e))  # pylint: disable=W0123
+                res.body = e[1].encode('utf-8')
+                if e[0] == 409:
+                    res.status = falcon.HTTP_409
+                    res.body = e[1].encode('utf-8')
+                elif e[0] == 408:
+                    res.status = falcon.HTTP_408
+                    res.body = e[1].encode('utf-8')
+                else:
+                    res.status = falcon.HTTP_500
```

Comment:
I understand that this hasn't changed, but I have the same concern as above. An "else" is not specific enough to determine whether this is actually a 500.

Reply:
See comment above.